rollout/trafficrouting/istio/istio.go

```diff
@@ -1406,7 +1406,7 @@ func splitManagedRoutesAndNonManagedRoutes(managedRoutes []v1alpha1.MangedRoutes
 	for _, route := range httpRoutes {
 		var found bool = false
 		for _, managedRoute := range managedRoutes {
-			if route.Name == managedRoute.Name {
+			if route["name"] == managedRoute.Name {
```

Comment:
As you are changing the signature to `map[string]interface{}` we don't have type safety as before. Maybe it is better to do type checks before doing `route["name"]`.

Reply:
I changed this slightly but the logic is actually dependent on the empty string return for the case when there is no name in the virtual service which is allowed via istio and argo rollouts (only in rollouts if it's a single route)